modules/apps/27-interchain-accounts/host/types/msgs.go

```diff
@@ -40,3 +43,35 @@ func (msg MsgUpdateParams) GetSigners() []sdk.AccAddress {
 
 	return []sdk.AccAddress{accAddr}
 }
+
+// NewMsgModuleQuerySafe creates a new MsgModuleQuerySafe instance
+func NewMsgModuleQuerySafe(signer string, requests []*QueryRequest) *MsgModuleQuerySafe {
+	return &MsgModuleQuerySafe{
+		Signer:   signer,
+		Requests: requests,
+	}
+}
+
+// ValidateBasic implements sdk.HasValidateBasic
+func (msg MsgModuleQuerySafe) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+	}
+
+	if len(msg.Requests) == 0 {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "no queries provided")
+	}
+
+	return nil
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgModuleQuerySafe) GetSigners() []sdk.AccAddress {
```

Comment:
is this needed since there's the signer annotation? cc. @srdtrk

Reply:
I think it is not needed for that reason

Reply:
I added it because the [removal of `GetSigners` is only on main](https://github.com/cosmos/ibc-go/pull/5305) and not in v8.x (`MsgUpdateParams` above also has it.)